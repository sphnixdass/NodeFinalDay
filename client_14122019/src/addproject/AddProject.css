legend
{
    text-align: center;
    font-size: large;
}

fieldset
{
    text-align: center;
}

.project
{
    width:25%;
    margin-left:15%;
}

.growl
{
    margin-top: 5%;
    margin-left: 10%;
    float:left;
}
