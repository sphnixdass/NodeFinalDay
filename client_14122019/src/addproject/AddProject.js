import React, {Component} from "react";
import axios from 'axios'
import {Fab, Grid, TextField,Select,MenuItem}from "@material-ui/core";
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import SaveIcon from '@material-ui/icons/Save';
import './AddProject.css'
import {Growl} from 'primereact/growl';
export class AddProject extends Component
{


    constructor(props)
    {
        super();
        this.state={
            projectId:0,
            projectName:"",
            location:"",
            countries:[]

        }
    }

    componentDidMount() {
        var countriesArr=[];
        axios.get('https://restcountries.eu/rest/v2/all').then(response=>{
            response.data.forEach(data=>{
               countriesArr.push(data.name);
            })
            this.setState({
                countries:countriesArr
            })
          console.log(this.state.countries);
        })

    }
    handle=input=>e=>{
        e.preventDefault();
        this.setState({[input]:e.target.value})
    }
   handleChange = event => {

       this.setState({location:event.target.value})
};
    save=(e)=>
    {
        e.preventDefault();
        console.log(this.state);

        var obj={
            "projectId":this.state.projectId,
            "projectName":this.state.projectName,
            "location":this.state.location
        }

        axios.post('http://localhost:5000/addProject/',obj)
            .then(res => {
               console.log(res);
                this.growl.show({severity: 'success', summary: 'Success Message', detail: 'Project created'});

            });
    }

    menuView(data, index){

        return (

            <MenuItem key={index} style={{ backgroundColor: '#a6cad8'}} value={data}>
            {data}
            </MenuItem>

    )
    }


    render()
    {

        return(
            <div>
            <div className="growl">
            <Growl ref={(el) => this.growl = el}  />
            </div>
    <form  onSubmit={this.save}  className="project" >
        <fieldset>
        <legend>Project Information</legend>
        <Grid container spacing={1}>
        <Grid item xs={12} sm={12}>
        <TextField id="projectId" type="number" label="Project Id" variant="outlined"  margin="normal"
        onChange={this.handle("projectId")}
        />
        </Grid>
        <Grid item xs={12} sm={12}>
        <TextField type="text" id="projectName" label="Project Name" variant="outlined"  margin="normal"
        onChange={this.handle("projectName")}
        />
        </Grid>
         <Grid item xs={12} sm={12}>
        <Select id="location" style={{left:2,width:210,top:10}} value={this.state.location} variant="outlined"
        onChange={this.handleChange}
        displayEmpty >

        {this.state.countries.map((country, i) => this.menuView(country, i))}
        </Select>
         </Grid>


        </Grid>
        <Fab style={{top:20}} color="primary" aria-label="add" type="submit" >
        <SaveIcon />
        </Fab>
        </fieldset>
        </form>
        </div>
        );
    }

}
