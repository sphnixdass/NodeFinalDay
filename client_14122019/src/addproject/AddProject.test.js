import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import {AddProject} from "./AddProject";
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
configure({adapter: new Adapter()});

import sinon from 'sinon';
it("should create an entry in component state", () => {
    const component = renderer.create(
        <AddProject/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    const component1 = shallow(<AddProject />);
    const form = component1.find('input');
    const legend = <legend>Project Information</legend>;
    // expect(wrapper.contains(welcome)).toBe(true);
    expect(component1.contains(legend)).toBe(true)
});
