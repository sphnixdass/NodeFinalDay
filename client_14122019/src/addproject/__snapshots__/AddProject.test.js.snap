// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create an entry in component state 1`] = `
<div>
  <div
    className="growl"
  >
    <div
      className="p-growl p-component p-growl-topright"
      id={null}
      style={null}
    >
      <div />
    </div>
  </div>
  <form
    className="project"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Project Information
      </legend>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="projectId"
              id="projectId-label"
            >
              Project Id
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="projectId"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-164 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-165"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="projectName"
              id="projectName-label"
            >
              Project Name
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="projectName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-164 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-165"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root"
            onClick={[Function]}
            style={
              Object {
                "left": 2,
                "top": 10,
                "width": 210,
              }
            }
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby=" location"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="location"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              onAnimationStart={[Function]}
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-164 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-165"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <button
        aria-label="add"
        className="MuiButtonBase-root MuiFab-root MuiFab-primary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "top": 20,
          }
        }
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </fieldset>
  </form>
</div>
`;
